google-ads-examples/src/main/java/com/google/ads/googleads/examples/shoppingads/AddPerformanceMaxRetailCampaign.java

```diff
@@ -0,0 +1,666 @@
+    private String countryCode;
+  }
+
+  public static void main(String[] args) throws IOException {
+    AddPerformanceMaxRetailCampaignParams params = new AddPerformanceMaxRetailCampaignParams();
+    if (!params.parseArguments(args)) {
+
+      // Either pass the required parameters for this example on the command line, or insert them
+      // into the code here. See the parameter class definition above for descriptions.
+      params.customerId = Long.parseLong("INSERT_CUSTOMER_ID_HERE");
+      params.merchantCenterAccountId = Long.parseLong("INSERT_MERCHANT_CENTER_ACCOUNT_ID_HERE");
+      params.countryCode = "INSERT_COUNTRY_CODE_HERE";
+    }
+
+    GoogleAdsClient googleAdsClient = null;
+    try {
+      googleAdsClient = GoogleAdsClient.newBuilder().fromPropertiesFile().build();
+    } catch (FileNotFoundException fnfe) {
+      System.err.printf(
+          "Failed to load GoogleAdsClient configuration from file. Exception: %s%n", fnfe);
+      System.exit(1);
+    } catch (IOException ioe) {
+      System.err.printf("Failed to create GoogleAdsClient. Exception: %s%n", ioe);
+      System.exit(1);
+    }
+
+    try {
+      new AddPerformanceMaxRetailCampaign()
+          .runExample(
+              googleAdsClient,
+              params.customerId,
+              params.merchantCenterAccountId,
+              params.countryCode);
+    } catch (GoogleAdsException gae) {
+      // GoogleAdsException is the base class for most exceptions thrown by an API request.
+      // Instances of this exception have a message and a GoogleAdsFailure that contains a
+      // collection of GoogleAdsErrors that indicate the underlying causes of the
+      // GoogleAdsException.
+      System.err.printf(
+          "Request ID %s failed due to GoogleAdsException. Underlying errors:%n",
+          gae.getRequestId());
+      int i = 0;
+      for (GoogleAdsError googleAdsError : gae.getGoogleAdsFailure().getErrorsList()) {
+        System.err.printf("  Error %d: %s%n", i++, googleAdsError);
+      }
+      System.exit(1);
+    }
+  }
+
+  // [START add_performance_max_retail_campaign]
+  /**
+   * Runs the example.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID.
+   * @param merchantCenterAccountId the Merchant Center account ID.
+   * @param countryCode sales country of products to include in the campaign.
+   */
+  private void runExample(
+      GoogleAdsClient googleAdsClient,
+      long customerId,
+      long merchantCenterAccountId,
+      String countryCode)
+      throws IOException {
+    // [START add_performance_max_retail_campaign_1]
+    // This campaign will override the customer conversion goals.
+    // Retrieve the current list of customer conversion goals.
+    List<CustomerConversionGoal> customerConversionGoals =
+        getCustomerConversionGoals(googleAdsClient, customerId);
+
+    // Performance Max campaigns require that repeated assets such as headlines
```

Comment:
I followed the golden here, but personally like this wording because it clarifies why some of the assets must be created beforehand, whereas some can be included in the generic mutate operation later on. In addition, folks may not know offhand that headlines and descriptions are repeated fields.